packages/react-query/src/createTRPCReact.tsx

```diff
@@ -53,8 +53,8 @@ export interface ProcedureUseQuery<
   TPath extends string,
 > {
   <
-    TQueryFnData = inferTransformedProcedureOutput<TProcedure>,
-    TData = inferTransformedProcedureOutput<TProcedure>,
+    TQueryFnData extends inferTransformedProcedureOutput<TProcedure> = inferTransformedProcedureOutput<TProcedure>,
+    TData extends inferTransformedProcedureOutput<TProcedure> = inferTransformedProcedureOutput<TProcedure>,
```

Comment:
@juliusmarminge  What's the difference between just assigning this compared to extending + assigning? 🤔

Reply:
@Nick-Lucas calls it the `extends =` pattern but it essentielly lets you have it as a generic (so others can override it) whilst keeping the constraint